Editor/AddressableImporter.cs

```diff
@@ -26,47 +26,81 @@ static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAsse
-                    if (entry != null)
-                    {
-                        entriesAdded.Add(entry);
-                        if (rule.HasLabel)
-                            Debug.LogFormat("[AddressableImporter] Entry created/updated for {0} with address {1} and labels {2}", assetPath, entry.address, string.Join(", ", entry.labels));
-                        else
-                            Debug.LogFormat("[AddressableImporter] Entry created/updated for {0} with address {1}", assetPath, entry.address);
-                    }
-                }
+                var group = AssetDatabase.LoadAssetAtPath<AddressableAssetGroup>(importedAsset);
+                if (group == null || group.IsDefaultGroup()|| group.entries.Count >= 1 || !settings.groups.Contains(group))
+                    continue;
+                
+                settings.RemoveGroup(group);
+                dirty = true;
             }
         }
-        if (entriesAdded.Count > 0)
+        foreach (var movedAsset in movedAssets)
         {
-            settings.SetDirty(AddressableAssetSettings.ModificationEvent.EntryMoved, entriesAdded, true);
-            AssetDatabase.SaveAssets();
-        }
-        if (importSettings.removeEmtpyGroups)
-        {
-            settings.groups.RemoveAll(_ => _.entries.Count == 0 && !_.IsDefaultGroup());
+            if (prefabStage == null || prefabStage.prefabAssetPath != movedAsset) // Ignore current editing prefab asset.
+                dirty |= ApplyImportRule(movedAsset, settings, importSettings);
         }
+        
+        if (dirty)
+            AssetDatabase.SaveAssets();
     }
 
     static AddressableAssetGroup CreateAssetGroup<SchemaType>(AddressableAssetSettings settings, string groupName)
     {
         return settings.CreateGroup(groupName, false, false, false, new List<AddressableAssetGroupSchema> { settings.DefaultGroup.Schemas[0] }, typeof(SchemaType));
     }
 
+    static bool ApplyImportRule(
+        string assetPath, 
+        AddressableAssetSettings settings,
+        AddressableImportSettings importSettings)
+    {
+        var dirty = false;
+        var matched = false;
+        foreach (var rule in importSettings.rules)
+        {
+            if (!rule.Match(assetPath))
+                continue;
+
+            var entry = CreateOrUpdateAddressableAssetEntry(settings, importSettings, rule, assetPath);
+            if (entry != null)
+            {
+                if (rule.HasLabel)
+                    Debug.LogFormat("[AddressableImporter] Entry created/updated for {0} with address {1} and labels {2}", assetPath, entry.address, string.Join(", ", entry.labels));
+                else
+                    Debug.LogFormat("[AddressableImporter] Entry created/updated for {0} with address {1}", assetPath, entry.address);
+            }
+
+            dirty = true;
+            matched = true;
+            break;
```

Comment:
I like this`stop when the first rule matched` logic since an asset is only allowed to be added to one group at a time. The previous implementation is the last rule win, which is weird.